src/tracer.js

```diff
@@ -104,14 +104,14 @@ export default class Tracer {
         this._metrics.spansStarted.increment(1);
         if (span.context().isSampled()) {
             this._metrics.spansSampled.increment(1);
-            if (rpcServer) {
+            if (!firstInProcess) {
                 this._metrics.tracesJoinedSampled.increment(1);
             } else if (!span.context().parentId) {
```

Comment:
change to hasParent which should return `this.parentId != null || this.parentIdStr != null`. Otherwise you may unnecessarily trigger Buffer construction